treeTable.js

```diff
@@ -237,21 +236,30 @@
     };
 
     TreeTable.prototype.redraw = function () {
+
+        if (this.collapsed.size === 0)
+        {
+            $.fn.dataTable.ext.search = $.fn.dataTable.ext.search.filter((it, i) => it.name !== "ttSearch");
```

Comment:
Not strictly necessary to do this, but if no rows are collapsed it's quicker to just delete all filters than to filter on an empty regex.

Reply:
Sorry, I don't really understand what's going on here. "ttSearch" is something built into datatable to hold the search filters?

Reply:
`ttSearch` is the name of our search filter - we filter out rows that are descendents of currently collapsed rows:  https://github.com/reside-ic/TreeTables/pull/10/files#diff-49985b7c735c1404bbb736fefa076a99R265

If there are no collapsed rows, we can just remove this filter and not replace it with anything else. Although in fact, we remove the old filter in either case (https://github.com/reside-ic/TreeTables/pull/10/files#diff-49985b7c735c1404bbb736fefa076a99R254) so could move this outside the if statement.

Reply:
Yes ok I see, so you always take out the old filter (by collapsed rows) and may or may not have to replace it with a new one. I think calling it "ttSearch" was what confused me, by making me think it was something to do with the search filter! Do you want to rearrange the code a bit so that the removal of the old filter always happens first, then it decides whether or not to return early? I don't really mind either way!

Reply:
Cool, I'll rename it to be clearer!